git-sync.php

```diff
@@ -107,12 +107,92 @@ public function init()
         $this->git = $this->controller->git;
     }
 
+    /**
+     * Execute command in background, cross-platform
+     * @param $cmd string Command to execute
+     * @return int|string Command result
+     */
+    public static function execInBackground($cmd)
+    {
+        if (false !== stripos(PHP_OS, 'win')) {
+            // Windows platform
+            return pclose(popen('start /B ' . $cmd, 'r'));
+        }
+        // Unix-like platform
+        return exec($cmd . ' > /dev/null &');
+    }
+
+    /**
+     * Checks if /bin/plugin exists in the given path
+     * @param string $path Path to check
+     * @return boolean
+     */
+    private static function doesBinPluginExist($path)
+    {
+        // Find /bin path
+        if (!($binPath = \realpath($path . DIRECTORY_SEPARATOR . 'bin'))) {
+            // /bin path could not be found automatically
+            return false;
+        }
+
+        // Find /plugin executable
+        $pluginPath = $binPath . DIRECTORY_SEPARATOR . 'plugin';
+        return \file_exists($pluginPath) && \is_readable($pluginPath);
+    }
+
+    /**
+     * Finds the root path of the running Grav instance
+     * @param array $config Git sync plugin config
+     * @return string Grav root path
+     */
+    private static function findRootPath($config = [])
+    {
+        // Find root path
+        if (\array_key_exists('root_path', $config)
+            && $rootPath = \realpath($config['root_path'])) {
+            return $rootPath;
+        }
+        // root path has not (or incorrectly) been set in configuration, guess automatically
+        if ($rootPath = \realpath(\dirname(__DIR__, 2))) {
+            return $rootPath;
+        }
+        // root path could not be found
+        throw new \RuntimeException('Git synchronize error: Grav root path is not configured properly.');
+    }
+    
+    /**
+     * Executes "sync" CLI command in background
+     * @param array $config Git sync plugin configuration
+     * @return bool Returns true when the command has been executed
+     * @throws \Exception
+     */
+    private function synchronizeInBackground($config = [])
+    {
+        $rootPath = static::findRootPath($config);
+        if (true !== static::doesBinPluginExist($rootPath)) {
+            throw new \RuntimeException(sprintf('/bin/plugin not found in %s, set path manually in Git sync plugin config', $rootPath));
+        }
+        // Execute CLI synchronize command in background
+        $cmd = 'cd ' . $rootPath . ' && php bin' . DIRECTORY_SEPARATOR . 'plugin git-sync sync';
+        static::execInBackground($cmd);
+
+        return true;
+    }
+
     public function synchronize()
     {
         if (!Helper::isGitInstalled() || !Helper::isGitInitialized()) {
             return true;
         }
 
+        // Synchronize in background ?
+        $pluginConfig = $this->config->get('plugins.' . $this->name);
+
+        if (isset($pluginConfig['execute_in_background']['enabled'])
```

Comment:
I think you should also add the following check:
```php
isset($pluginConfig['execute_in_background'])
```
For the case the config file does not have this entry.

Reply:
👍